org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java

```diff
@@ -101,6 +101,12 @@ public static void initialize() {
 		defEclipsePrefs.put(StubUtility.CODEGEN_EXCEPTION_VAR_NAME, "e"); //$NON-NLS-1$
 		defEclipsePrefs.put(StubUtility.CODEGEN_ADD_COMMENTS, Boolean.FALSE.toString());
 
+		defEclipsePrefs.put("recommenders.chain.min_chain_length", "2");
+		defEclipsePrefs.put("recommenders.chain.max_chain_length", "4");
```

Comment:
What are your thoughts about `4` here. Should it be `3` ? I really don't think we offer much insight at that level. Most of the time the interesting API is maybe 2 invocations away.

Reply:
Should we start with `max` as `2` and see the feedback ?